### PR TITLE
Update IP assignment guidance

### PR DESCRIPTION
**Pull request by xw-g**
* Add a WARNING that users who run the test own the responsibility of not leaking test traffic to internet.
* Add `100.64.0.0/10`, a non-globally routable CGN range, which usually is used inside service providers. https://www.iana.org/go/rfc6598.
* Correct the name of `198.18.0.0/15` to `Device Benchmark Testing` (RFC 2544).
* Removed other "arbitrary" IP ranages, we'd like to deprecate them from the tests.
* Update IP check script accordingly.

Formatting:
* Simplify the wording in the `IP Addresses Assignment` section.

### CODE REVIEW

File: tools/check_ip_addresses.pl [hunk 1]
```diff
@@ -43,7 +43,11 @@ END
   next if $ip =~ /192\.0\.2\./;         # TEST-NET-1 (RFC 5737)
   next if $ip =~ /198\.51\.100\./;      # TEST-NET-2 (RFC 5737)
```
Comment (morrowc) [inline tools/check_ip_addresses.pl:44]: is there a reason to do this with regex and not to just:
```perl
  candidate = ipaddres.IPAddress($ip)
  result = false
  for net in netsDict {
      if candidate in net {
       result=true
       break
     }
  }

if !result errorBdaly
```

**Review by robshakir — APPROVED** [review]

File: tools/check_ip_addresses.pl (hunk 1 above)
Comment (xw-g) [reply tools/check_ip_addresses.pl:44]: @robshakir you might know?  (does not block this PR merge)

Comment (robshakir) [reply tools/check_ip_addresses.pl:44]: I am not sure who wrote this perl originally -- honestly, if we refactor it, should we just make it not-perl? :-)

**Review by robshakir — APPROVED** [review]
